core/src/main/java/com/novoda/merlin/MerlinsBeard.java

```diff
@@ -10,6 +10,20 @@
  */
 public class MerlinsBeard {
 
+    public enum NetworkType {
```

Comment:
it's a series of articles. One reference is this one https://medium.com/google-developers/developing-for-android-ii-bb9a51f8c8b9 the point is the memory overhead that you are enforcing every user of the library to have.  This is the reason the Android team created the annotation type checking.  

if you create `isConnectedOnWifi()` internally we can just use `ConnectivityManager.TYPE_WIFI` for now